### ISSUE TITLE
The implemented actor critic algorithm is indeed REINFORCE-with-baseline algorithm

### ISSUE THREAD

**Issue by SunHaozhe** [pytorch/examples#573]
The algorithm implemented in `examples/reinforcement_learning/actor_critic.py` is indeed REINFORCE-with-baseline algorithm, but not actor-critic algorithm, because its learned state-value function is used only as a baseline, not as a critic, i.e. it is not used for bootstrapping (updating the value estimate for a state from the estimated values of subsequent states) [1].

According to [1], REINFORCE-with-baseline algorithm is not considered to be an actor-critic method.

[1] Richard S. Sutton and Andrew G. Barto. Reinforcement Learning: An Introduction. The
MIT Press, second edition, 2018.

**Comment by Hoiy** [comment]
Agree with @SunHaozhe

Adding some more comment (correct me if I am wrong):

1. Actor-Critic algorithm should be able to update parameters at every time step (not every end of the episode) to make it converge faster due to bootstrapping.
2. Actor and Critic should never share parameters.
3. Should not standardize the output of critic as they should be the weight of actor's loss function, which is carefully designed.
4. The loss function of critic should not be some random distance metric and should be carefully designed such that the gradient is the same as the pseudo code.

**Comment by rodrigodesalvobraz** [comment]
Indeed. The relevant explanation is on page 331.

**Comment by msaroufim** [comment]
Closing since this a dup of #744
